### DISCUSSION TITLE
Efficient Ways to Move Large Data from ClickHouse to SQL Server or Other Databases

### DISCUSSION THREAD

**Discussion by anuragladdha-ml** [ClickHouse/ClickHouse#64178] [Q&A]
Hey, I'm looking for efficient ways to transfer a large dataset from ClickHouse to SQL Server or any other database. The dataset is relatively large, consisting of over 10 million rows and 100 columns.
I've explored a few options so far:

Pandas: While Pandas is a powerful data manipulation library in Python, it seems to be relatively slow for transferring such a large dataset due to its single-threaded nature.
Dask: I attempted to use Dask, a parallel computing library for Python, but encountered difficulties in establishing a connection between ClickHouse and Dask.

I'm seeking recommendations on alternative methods or tools that could facilitate this data transfer more efficiently, especially for datasets of this size or larger.
Are there any other approaches, libraries, or techniques you would suggest for effectively moving large datasets from ClickHouse to SQL Server or other database systems? I'm open to solutions in Python, SQL, or any other language or tool that can handle this task efficiently.
Please provide your insights, and if you have experience with similar data transfer scenarios, I'd appreciate hearing about your approach and any potential pitfalls to watch out for.
Thank you in advance for your help!

**Comment by yiyang-shao** [comment]
Hi! 

Firstly, you can try dump clickhouse data into an intermediate file(e.g. [Parquet](https://clickhouse.com/docs/en/integrations/data-formats/parquet#exporting-to-parquet-format)) and then insert the file into the destination database.

Also, you may consider some third-party solutions such as [airbyte](https://airbyte.com/) or [seatunnel](https://seatunnel.apache.org/docs/about).
